cmd/channel/main.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"github.com/cloudevents/sdk-go/v1/cloudevents"
+	"knative.dev/eventing-kafka/pkg/channel/channel"
+	"knative.dev/eventing-kafka/pkg/channel/constants"
+	"knative.dev/eventing-kafka/pkg/channel/env"
+	channelhealth "knative.dev/eventing-kafka/pkg/channel/health"
+	"knative.dev/eventing-kafka/pkg/channel/producer"
+	commonk8s "knative.dev/eventing-kafka/pkg/common/k8s"
+	kafkautil "knative.dev/eventing-kafka/pkg/common/kafka/util"
+	"knative.dev/eventing-kafka/pkg/common/prometheus"
+	"go.uber.org/zap"
+	eventingChannel "knative.dev/eventing/pkg/channel"
+	"knative.dev/eventing/pkg/kncloudevents"
+	"knative.dev/pkg/logging"
+)
+
+// Variables
+var (
+	logger     *zap.Logger
+	masterURL  = flag.String("masterurl", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
+	kubeconfig = flag.String("kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
+)
+
+// The Main Function (Go Command)
+func main() {
+
+	// Parse The Flags For Local Development Usage
+	flag.Parse()
+
+	// Initialize A Knative Injection Lite Context (K8S Client & Logger)
+	ctx := commonk8s.LoggingContext(context.Background(), constants.Component, *masterURL, *kubeconfig)
+
+	// Get The Logger From The Context & Defer Flushing Any Buffered Log Entries On Exit
+	logger = logging.FromContext(ctx).Desugar()
+	defer func() { _ = logger.Sync() }()
+
+	// Load Environment Variables
+	environment, err := env.GetEnvironment(logger)
+	if err != nil {
+		logger.Fatal("Invalid / Missing Environment Variables - Terminating")
+		return // Quiet The Compiler ; )
```

Comment:
The compiler was issuing warnings to the effect that the `environment` variable might be used below since it did not detect the Fatal() call as terminating execution.  Thus, to quiet the compiler we had added the return and comment.  

After looking at the other use cases the only difference I could spot was the missing `zap.Error(err)` in the Fatal() call.  Sure enough - adding that allowed the compiler to detect the end of execution flow and not flag the subsequent usages of `environment`.

Obviously that fix is better (and consistent) so we'll switch over/back to that - thanks for making me take a second look at it - learned something ; )